include/fiction/algorithms/physical_design/wiring_reduction.hpp

```diff
@@ -0,0 +1,886 @@
+                                        {
+                                            for (uint64_t o = 0; o < offset; ++o)
+                                            {
+                                                offset_offset++;
+                                                if ((fanin.x > 0) &&
+                                                    (offset_matrix[fanin.y][fanin.x] !=
+                                                     offset_matrix[fanin.y][fanin.x - 1]) &&
+                                                    (layout_copy.incoming_data_flow(fanin)[0].x != fanin.x))
+                                                {
+                                                    fanin = {fanin.x - 1, fanin.y, fanin.z};
+                                                }
+                                                else
+                                                {
+                                                    fanin = layout_copy.incoming_data_flow(fanin)[0];
+                                                    break;
+                                                }
+                                            }
+                                        }
+
+                                        // Create signals for the new coordinates
+                                        signals.push_back(lyt.make_signal(
+                                            lyt.get_node({fanin.x - offset + offset_offset, fanin.y, fanin.z})));
+                                    }
+                                }
+                            });
+
+                        // Move the node to the new coordinates
+                        lyt.move_node(lyt.get_node(old_coord), new_coord, signals);
+                    }
+                }
+            }
+        }
+    }
+
+    // Calculate bounding box for optimized layout size
+    const auto bounding_box            = bounding_box_2d(lyt);
+    const auto optimized_layout_width  = bounding_box.get_x_size();
+    const auto optimized_layout_height = bounding_box.get_y_size();
+
+    // Resize the layout to the optimized size
+    lyt.resize({optimized_layout_width, optimized_layout_height, lyt.z()});
+}
+
+}  // namespace detail
+
+/**
+ * Perform wiring reduction on a 2DDWave-clocked Cartesian gate-level layout.
+ *
+ * This algorithm performs wiring reduction of a 2DDWave-clocked Cartesian gate-level layout by iteratively
+ * finding tiles with excess wiring that can be deleted without changing the correctness of the underlying function.
+ * The process continues until no further wires can be deleted.
+ *
+ * @tparam Lyt Cartesian gate-level layout type.
+ * @param lyt The 2DDWave-clocked layout whose wiring is to be reduced.
+ * @param pst Pointer to a `wiring_reduction_stats` object to record runtime statistics (optional).
+ */
+template <typename Lyt>
+void wiring_reduction(const Lyt& lyt, wiring_reduction_stats* pst = nullptr) noexcept
```

Comment:
This function seems to do a lot itself instead of being just an entry point for calling the `*_impl` functor. Can we refactor this behavior or is it important for testing?